splunklogger.js

```diff
@@ -451,17 +629,20 @@ SplunkLogger.prototype._sendEvents = function(context, callback) {
  * @throws Will throw an error if the <code>context</code> parameter is malformed.
  * @public
  */
-SplunkLogger.prototype.send = function (context, callback) {
+SplunkLogger.prototype.send = function(context, callback) {
     callback = callback || function(){};
     context = this._initializeContext(context);
     
+    // Store the context, and its estimated length
     this.contextQueue.push(context);
+    this.eventSizes.push(Buffer.byteLength(JSON.stringify(this._makeBody(context)), "utf8"));
 
-    if (context.config.autoFlush) {
-        this.flush(callback);
-    }
-    else {
-        callback();
+    var batchOverSize = this.calculateBatchSize() > context.config.maxBatchSize;
```

Comment:
I'm not sure I see why you need to calculate this every time. Namely, why do you need to keep a list of the batch sizes and not just keep the total batch size up until now?